python/eups/Eups.py

```diff
@@ -3041,10 +3041,32 @@ def getDependentProducts(self, topProduct, setup=False, shouldRaise=False,
         if not prodtbl:
             return dependentProducts
 
-        for product, optional, recursionDepth in prodtbl.dependencies(self, recursive=True, recursionDepth=1,
-                                                                      followExact=followExact,
-                                                                      productDictionary=productDictionary,
-                                                                      requiredVersions=requiredVersions):
+        productDict = {}
+        dependencies = prodtbl.dependencies(self, recursive=True, recursionDepth=1,
```

Comment:
Calling `dependencies()` three times seems pretty wasteful.  The `defaultDeps` shouldn't be changing all the time, should they?  Can't they be cached or otherwise stored, enabling a reduction to just one `dependencies()` call (with `addDefaultProduct` set to the proper value)?